src/haskell/parser/src/KoreLexemeImpl.hs

```diff
@@ -0,0 +1,244 @@
+module KoreLexemeImpl where
+
+import qualified CharDict
+import           CharSet
+import           CString
+import           KoreAST
+
+import           Control.Applicative              ((<|>))
+import           Control.Monad                    (void)
+import           Data.Attoparsec.ByteString.Char8 (Parser)
+import qualified Data.Attoparsec.ByteString.Char8 as Parser
+import qualified Data.Attoparsec.ByteString       as BParser (runScanner)
+import qualified Data.ByteString.Char8            as Char8
+import           Data.Char                        (isHexDigit, isOctDigit)
+
+idParser :: Parser Id
+idParser = lexeme idRawParser
+
+metaIdParser :: Parser MetaId
+metaIdParser = lexeme metaIdRawParser
+
+stringLiteralParser :: Parser StringLiteral
+stringLiteralParser = lexeme stringLiteralRawParser
+
+moduleNameParser :: Parser ModuleName
+moduleNameParser = lexeme moduleNameRawParser
+
+data CommentScannerState = COMMENT | STAR | END
+
+multiLineCommentToken :: Parser ()
+multiLineCommentToken = do
+--    void (Parser.string (Char8.pack "/*"))
+    (_,state) <- BParser.runScanner COMMENT delta'
+    case state of
+        END -> return ()
+        _ -> fail "Unfinished comment."
+  where
+    delta' s c = delta s (toEnum (fromEnum c))
+    delta END _    = Nothing
+    delta _ '*'    = Just STAR
+    delta STAR '/' = Just END
+    delta _ _      = Just COMMENT
+
+singleLineCommentToken :: Parser ()
+singleLineCommentToken = do
+--    void (Parser.string (Char8.pack "//"))
+    void (Parser.scan COMMENT delta)
+  where
+    delta END _  = Nothing
+    delta _ '\n' = Just END
+    delta _ _    = Just COMMENT
+
+
+spaceChars :: [Char]
+spaceChars = [' ', '\t', '\n', '\v', '\f', '\r']
+
+whitespaceChunk :: Parser ()
+whitespaceChunk =
+    prefixBasedParsersWithDefault
+        stringParser
+        (return ())
+        (
+            [ ("/*", multiLineCommentToken *> whitespaceChunk)
+            , ("//", singleLineCommentToken *> whitespaceChunk)
+            ]
+            ++
+            map (\c -> ([c], Parser.skipSpace *> whitespaceChunk)) spaceChars
+        )
+
+-- TODO: Rewrite this, or parts of this, using Parser.scan
+skipWhitespace :: Parser ()
+skipWhitespace = whitespaceChunk
+
+lexeme :: Parser a -> Parser a
+lexeme p = p <* skipWhitespace
+
+genericIdParser :: CharSet -> CharSet -> (String -> a) -> Parser a
+genericIdParser firstCharSet charSet constructor = do
+    c <- Parser.peekChar'
+    idChar <- if not (c `CharSet.elem` firstCharSet)
+        then fail ("genericidParser: Invalid first character '" ++ c : "'.")
```

Comment:
Maybe this is a bit off, but have you considered providing context information when parsing fails? It does not look like there is any support for that in this parser.